### PR TITLE
EN-4934: unit tests for sync package

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: process/sync/metablock_test.go [hunk 1]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+	shardCoordinator := mock.NewOneShardCoordinatorMock()
+	account := &mock.AccountsStub{}
+
+	uint64Converter := uint64ByteSlice.NewBigEndianConverter()
+	nonceToByteSlice := uint64Converter.ToByteSlice(uint64(0))
+
+	store := createStore()
+	errKeyNotFound := errors.New("key not found")
+	hash := []byte("hash")
+	//buff := []byte("buff")
```
Comment (miiu96) [inline process/sync/metablock_test.go]: remove commented line

File: process/sync/metablock_test.go [hunk 2]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+
+		return nil
+	}
+
+	store.GetCalled = func(unitType dataRetriever.UnitType, key []byte) ([]byte, error) {
+		if unitType == dataRetriever.ShardHdrNonceHashDataUnit {
+			if bytes.Equal(key, nonceToByteSlice) {
+				return nil, errKeyNotFound
+			}
+			return []byte("hash"), nil
```
Comment (miiu96) [inline process/sync/metablock_test.go]: use hash variable here

File: process/sync/metablock_test.go [hunk 3]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+	blockWithLastNotarized := make(map[uint32]uint64, 0)
+	blockWithFinalNotarized := make(map[uint32]uint64, 0)
+
+	lastNotarized[0] = 1
+	finalNotarized[0] = 1
+	blockWithLastNotarized[0] = 1
+	blockWithFinalNotarized[0] = 1
+	startNonce := uint64(0)
+
+	shardData := block.ShardData{TxCount: 10, HeaderHash: []byte("hash")}
```
Comment (miiu96) [inline process/sync/metablock_test.go]: also here

File: process/sync/metablock_test.go [hunk 4]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+	shardCoordinator := mock.NewOneShardCoordinatorMock()
+	account := &mock.AccountsStub{}
+
+	uint64Converter := uint64ByteSlice.NewBigEndianConverter()
+	nonceToByteSlice := uint64Converter.ToByteSlice(uint64(0))
+
+	store := createStore()
+	errKeyNotFound := errors.New("key not found")
+	hash := []byte("hash")
+	//buff := []byte("buff")
```
Comment (miiu96) [inline process/sync/metablock_test.go]: remove commented code

File: process/sync/metablock_test.go [hunk 5]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+
+		return nil
+	}
+
+	store.GetCalled = func(unitType dataRetriever.UnitType, key []byte) ([]byte, error) {
+		if unitType == dataRetriever.ShardHdrNonceHashDataUnit {
+			if bytes.Equal(key, nonceToByteSlice) {
+				return nil, errKeyNotFound
+			}
+			return []byte("hash"), nil
```
Comment (miiu96) [inline process/sync/metablock_test.go]: use hash variable

File: process/sync/metablock_test.go [hunk 6]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+	shardCoordinator := mock.NewOneShardCoordinatorMock()
+	account := &mock.AccountsStub{}
+
+	uint64Converter := uint64ByteSlice.NewBigEndianConverter()
+	nonceToByteSlice := uint64Converter.ToByteSlice(uint64(0))
+
+	store := createStore()
+	errKeyNotFound := errors.New("key not found")
+	hash := []byte("hash")
+	//buff := []byte("buff")
```
Comment (miiu96) [inline process/sync/metablock_test.go]: also here

File: process/sync/metablock_test.go [hunk 7]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+
+		return nil
+	}
+
+	store.GetCalled = func(unitType dataRetriever.UnitType, key []byte) ([]byte, error) {
+		if unitType == dataRetriever.ShardHdrNonceHashDataUnit {
+			if bytes.Equal(key, nonceToByteSlice) {
+				return nil, errKeyNotFound
+			}
+			return []byte("hash"), nil
```
Comment (miiu96) [inline process/sync/metablock_test.go]: use hash variable

File: process/sync/metablock_test.go [hunk 8]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+	t.Parallel()
+
+	pools := createMockMetaPools()
+	blkc := initBlockchain()
+	rnd := &mock.RounderMock{}
+	blkExec := &mock.BlockProcessorMock{
+		AddLastNotarizedHdrCalled: func(shardId uint32, processedHdr data.HeaderHandler) {},
+	}
+	hasher := &mock.HasherMock{}
+	marshalizer := &mock.MarshalizerStub{
```
Comment (miiu96) [inline process/sync/metablock_test.go]: maybe you can use MarshalizerMock

File: process/sync/metablock_test.go [hunk 9]
```diff
@@ -2649,6 +2651,290 @@ func TestMetaBootstrap_ApplyNotarizedBlockShouldWork(t *testing.T) {
...
+		&mock.NetworkConnectionWatcherStub{},
+	)
+
+	lastNotarized := make(map[uint32]uint64, 0)
+	finalNotarized := make(map[uint32]uint64, 0)
+	blockWithLastNotarized := make(map[uint32]uint64, 0)
+	blockWithFinalNotarized := make(map[uint32]uint64, 0)
+
+	lastNotarized[0] = 0
+
```
Comment (miiu96) [inline process/sync/metablock_test.go]: remove empty line

File: process/sync/shardblock_test.go [hunk 10]
```diff
@@ -5104,3 +5106,206 @@ func TestShardBootstrap_SetStatusHandlerNilHandlerShouldErr(t *testing.T) {
...
+	pools.HeadersCalled = func() storage.Cacher {
+		sds := &mock.CacherStub{}
+		sds.RegisterHandlerCalled = func(func(key []byte)) {
+		}
+
+		return sds
+	}
+	pools.HeadersNoncesCalled = func() dataRetriever.Uint64SyncMapCacher {
+		hnc := &mock.Uint64SyncMapCacherStub{}
+		hnc.RegisterHandlerCalled = func(handler func(nonce uint64, shardId uint32, hash []byte)) {}
```
Comment (miiu96) [inline process/sync/shardblock_test.go]: put second bracket on a new line

File: process/sync/metablock_test.go (hunk 4 above)
Comment (sasurobert) [inline process/sync/metablock_test.go]: delete commented line

File: process/sync/metablock_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 4 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 3 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 5 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 6 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 7 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 8 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 9 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/shardblock_test.go (hunk 10 above)
Comment (bogdan-rosianu) [reply process/sync/shardblock_test.go]: done

File: process/sync/metablock_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

File: process/sync/metablock_test.go (hunk 4 above)
Comment (bogdan-rosianu) [reply process/sync/metablock_test.go]: done

**Pull request merged**